enhancements/proxy/global-cluster-egress-proxy.md

```diff
@@ -0,0 +1,285 @@
+so they will trust my proxy.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+This enhancement introduces a cluster scoped proxy configuration resource.
+The resource includes fields to:
+
+* specify an https proxy url
+* specify an http proxy url
+* specify additional domains that should not be proxied, in addition to some system defined ones
+* specify a reference to a user defined configmap containing additional CAs that should be trusted when
+connecting to the proxy.
+* specify endpoints that can be used to validate the proxy configuration is functional
+
+All of this information with the exception of the validation endpoints can be provided
+at install time to ensure that a cluster can bootstrap successfully even if it needs to
+reach external services via the proxy to do so.
+
+The information can also be modified at runtime.  If it is modified at runtime, a controller
+will confirm that the validation endpoints can be successfully reached using the new configuration
+before accepting the new configuration.  Once accepted, the configuration is moved into the status
+section of the proxy config resource.  Components should only consume the proxy configuration from
+this location.  
+
+Similarly, any user provided CAs will only be copied into an "accepted" CAs 
+configmap after confirming the validation endpoints can be accessed using the new CAs.
+
+Additional behaviors:
+
+* configmaps labeled with `config.openshift.io/inject-trusted-cabundle: "true"` will have the current 
+set of additional CAs injected into them by logic in the cluster network operator.
+* deployments with the `config.openshift.io/inject-proxy: <container-name>` will get the current proxy
+environment variables injected (HTTP_PROXY, HTTPS_PROXY, NO_PROXY) by the cluster version operator.
+
+Critical touch points for the administrator:
+
+* edits cluster scoped proxy config resource (spec fields)
+* provides a configmap of additional CAs
+
+Critical touch points for proxy configuration consumers:
+
+* Operator consumes status fields from cluster scoped proxy config resource and updates its operand accordingly
+* *May* consume the "accepted CAs" configmap (openshift-config-managed/trusted-ca-bundle) to get CAs
+* *Should* create their own configmap with the `config.openshift.io/inject-trusted-cabundle: "true"` label and
+consume the CA bundle from there.
+* Operator deployment may request proxy environment injection via the `config.openshift.io/inject-proxy: <container-name>`
+annotation since operators cannot control their own environment variables, but the oprator is responsible for mounting
+a configmap to pick up the CAs if it needs them.
+
+Automatic no proxy behavior:
+
+The following domains/hosts are automatically added to the no proxy configuration:
+* TODO
+
+Known limitations/future enhancements:
+
+* The no proxy entries that are automatically added do not include the cluster domain generically, so
```

Comment:
What addressing the ingress domain?

Reply:
we aren't proxying the ingress domain, right?

but what i was trying to cover here was the fact that we do not add "*.mycluster.com" to the noproxy list.

so yes, maybe i should have said ingress domain?

Reply:
switched to ingress domain.